nixos/modules/services/system/guix-daemon.nix

```diff
@@ -0,0 +1,72 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}: let
+  cfg = config.guix;
+in {
+  options = with lib; {
+    guix = {
+      enable = mkEnableOption "guix";
+
+      package = mkOption {
+        type = with types; package;
+        default = pkgs.guix;
+        defaultText = literalExpression "pkgs.guix";
+        description = mdDoc ''
+          This option specifies the Guix package to be used in the system.
+        '';
+      };
+
+      substituteUrls = mkOption {
+        type = with types; listOf str;
+        default = [
+          "https://ci.guix.gnu.org"
+          "https://bordeaux.guix.gnu.org"
+        ];
+        description = mdDoc ''
+          List of urls to be used as substituters.
+        '';
+      };
+
+      cores = mkOption {
+        type = with types; int;
+        default = 0;
+        description = mdDoc ''
+          Number of CPU cores to build each derivation, 0 means as many as available.
+        '';
+      };
+
+      maxJobs = mkOption {
+        type = with types; int;
+        default = 1;
+        description = mdDoc ''
+          Allow at most n build jobs in parallel. The default value is 1. Setting it to 0 means that no builds will be performed locally; instead, the daemon will offload builds (see Using the Offload Facility), or simply fail.
+        '';
+      };
+
+      extraArgs = mkOption {
+        type = with types; listOf (coercedTo anything (x: "${x}") str);
```

Comment:
This type is a bit weird. I see what you're going for with it being easier to write ints and so on, but I think there are better ways to do this. It doesn't make sense to put bools, attrsets and probably not lists either here, unless guix can read nix syntax in its arguments. `anything` will also accept `types.function`, which errors out on string interpolation.

Also, AFAIK string interpolation just invokes the same logic that is used in `toString`, meaning `(x: "${x}")` can be replaced with `toString`.

Maybe we could just keep it to `listOf str`? Alternatively you could look into using `lib.cli.toGNUCommandLineShell`.